great_expectations/core/partitioners.py

```diff
@@ -34,24 +40,36 @@ class PartitionerDividedInteger(pydantic.BaseModel):
     ] = "partition_on_divided_integer"
 
 
+@public_api
 class PartitionerModInteger(pydantic.BaseModel):
     mod: int
     column_name: str
     method_name: Literal["partition_on_mod_integer"] = "partition_on_mod_integer"
 
 
+@public_api
 class PartitionerColumnValue(pydantic.BaseModel):
     column_name: str
     method_name: Literal["partition_on_column_value"] = "partition_on_column_value"
 
 
+@public_api
 class PartitionerMultiColumnValue(pydantic.BaseModel):
     column_names: List[str]
     method_name: Literal[
         "partition_on_multi_column_values"
     ] = "partition_on_multi_column_values"
 
 
+@public_api
+class PartitionerConvertedDatetime(pydantic.BaseModel):
+    column_name: str
+    method_name: Literal[
+        "partition_on_converted_datetime"
+    ] = "partition_on_converted_datetime"
+    date_format_string: str
```

Comment:
Should we do any validation on this string?

Reply:
[current behavior](https://github.com/great-expectations/great_expectations/blob/2bb6081895c1979dfcc74687f88a3b2ff0944978/great_expectations/datasource/fluent/sqlite_datasource.py#L51) is to not validate the string, so i think it's likely okay to not. I'm not 100% how we would validate it, either, since its correctness depends on the shape of the source data.